pynest/nest/tests/test_pp_psc_delta.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+#
+# test_pp_psc_delta.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+#
+# tests for pp_psc_delta
+#
+
+import unittest
+import nest
+
+
+# 1) check for reasonable firing rate
+# 2) check if fixed dead-time is respected
+
+
+class PpPscDeltaTestCase(unittest.TestCase):
+
+    def test_rate_and_fixed_dead_time(self):
+        # test parameters
+        d = 25.0
+        lam = 10.0
+        T = 10000.0
+
+        nest.ResetKernel()
+        nrn = nest.Create('pp_psc_delta')
+
+        params = {'tau_m': 10.0,
+                  'C_m': 250.0,
+                  'dead_time': d,
+                  'dead_time_random': False,
+                  'dead_time_shape': 1,
+                  'with_reset': False,
+                  'tau_sfa': 34.0,
+                  'q_sfa': 0.0,  # // mV, reasonable default is 7 mV
+                  'c_1': 0.0,
+                  'c_2': lam,
+                  'c_3': 0.25,
+                  'I_e': 0.0,
+                  't_ref_remaining': 0.0
+                  }
+
+        nest.SetStatus(nrn, params)
+
+        sd = nest.Create('spike_detector')
+        nest.Connect(nrn, sd)
+        nest.Simulate(T)
+
+        spikes = nest.GetStatus(sd)[0]['events']['times']
+        rate_sim = len(spikes) / (T * 1e-3)
+        rate_ana = 1. / (1. / lam + d * 1e-3)
+        ratio = rate_sim / rate_ana
+
+        # This could fail due to bad luck. However, if it passes once,
+        # then it should always do so, since the random numbers are
+        # reproducible in NEST.
+        self.assertLess(0.5, ratio)
+        self.assertLess(ratio, 1.5)
+
+        isi = []
+        for i in xrange(1, len(spikes)):
```

Comment:
`xrange` disappears in Python 3. A much more efficient way to do this would be using numpy arrays.